ADALiOS/ADALiOS/ADTokenCacheStoreItem+Internal.m

```diff
@@ -30,37 +30,38 @@ @implementation ADTokenCacheStoreItem (Internal)
 
 - (void)fillExpiration:(NSDictionary*)responseDictionary
 {
-    id expires_in = [responseDictionary objectForKey:@"expires_on"];
-    if (!expires_in)
+    id expires_in = [responseDictionary objectForKey:OAUTH2_EXPIRES_IN];
+    
+    //if access_token was not return then ONLY use id_token_expires_in
+    if (![NSString adIsStringNilOrBlank:[responseDictionary objectForKey:OAUTH2_ACCESS_TOKEN]])
```

Comment:
if there is no access token returned then only we use OAUTH2_ID_TOKEN_EXPIRES_IN (returned all the time) in the response. access_token is paired with expires_in and id_token is paired with id_token_expires_in. I cannot check if id_token exists because it is returned in all the responses. It is the lack of access token that matters.